src/features/countries/countriesSlice.js

```diff
@@ -0,0 +1,68 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const URL = 'https://countriesnow.space/api/v0.1/countries/population';
+const initialState = {
+  countries: [],
+  isLoading: true,
+  error: '',
+};
+
+export const fetchCountries = createAsyncThunk('countries/fetchCountries', async () => {
+  try {
+    const response = await fetch(URL);
+
+    if (!response.ok) {
+      throw new Error('Failed to fetch countries');
+    }
+
+    const data = await response.json();
+    return data.data;
+  } catch (error) {
+    throw new Error(`Failed to fetch the data, ${error}`);
+  }
+});
+
+const getRecentPopulation = (populationCounts) => {
+  const size = populationCounts.length - 1;
+  return populationCounts[size].value;
+};
+
+const countriesSlice = createSlice({
+  name: 'countries',
+  initialState,
+  reducers: {
+    setFetchedCountries: (state, action) => {
+      state.countries = action.payload;
+      state.isLoading = false;
+      state.error = '';
+    },
+  },
+  extraReducers(builder) {
+    builder
+      .addCase(fetchCountries.pending, (state) => {
+        state.isLoading = true;
+      })
+      .addCase(fetchCountries.fulfilled, (state, action) => {
+        state.isLoading = false;
+        const data = action.payload.map((item) => ({
+          country: item.country,
+          iso3: item.iso3,
+          code: item.code,
+          populationCounts: getRecentPopulation(item.populationCounts),
+          demography: item.populationCounts,
+        }));
+        state.countries = data;
+      })
+      .addCase(fetchCountries.rejected, (state, action) => {
+        state.countries = [];
+        state.error = action.payload;
+      });
+  },
+});
+
+/* eslint-disable max-len */
+export const getCountryByIso3 = (state, iso3) => state.countries.countries.find((country) => country.iso3 === iso3);
+/* eslint-disable max-len */
```

Comment:
- [ ] Kindly do not disable linters, fixing linters helps us to write strong codebase 🙃